components/chapters/LoginModal.tsx

```diff
@@ -0,0 +1,177 @@
+        setUser(user);
+        setLoaded(true);
+      } else {
+        alert('Invalid private key')
+      }
+    }
+  }
+
+  function checkValidKey() {
+    if (userPrivateKey.length !== 64) return false;
+
+    const user = window.localStorage.getItem('user');
+    if (user) {
+      const userObj = JSON.parse(user);
+      const privateKey = Secp256k1.uint256(Buffer.from(userPrivateKey, 'hex'), 16)
+      const publicKey = Secp256k1.generatePublicKeyFromPrivateKeyData(privateKey);
+
+      if (userObj.publicKey.x === publicKey.x && userObj.publicKey.y === publicKey.y) {
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  function clearProgress() {
+    window.localStorage.removeItem('user');
+    window.localStorage.removeItem('loggedIn');
+
+    props.onClose();
+  }
+
+  function copy() {
+    navigator.clipboard.writeText(userPrivateKey)
+    setCopied(true)
+
+    setTimeout(() => {
+      setCopied(false)
+    }, 2000)
+  }
+
+  function signOut() {
+    setLoaded(false);
+    props.onLogout();
+    props.onClose();
+  }
+
+  return (
+    <Modal
+      isOpen={props.open}
+      className="font-nunito top-1/2 left-1/2 outline-none p-5 transform -translate-x-1/2 -translate-y-1/2 absolute w-[32rem] rounded-lg shadow-lg bg-[var(--back)] text-white"
+      contentLabel="Login Modal"
+    >
+      <div className="float-right flex justify-end">
+        <button onClick={props.onClose} aria-label="Close">
+          <Close className="h-6 w-6" />
+        </button>
+      </div>
+
+      <div className="p-4">
+      {!loaded && !props.signedIn && (
+        <>
+          <Avatar avatar={1} size={80} />
+          <h2 className="mb-4 mt-5 text-3xl font-bold">Load your progress</h2>
+          <p className="mb-5">
+            Enter your personal code below to restore your chapter and challenge progress, as well as your avatar. <a href="#" className='text-white underline'>No code yet?</a>
+          </p>
+
+          <div className="flex flex-col justify-between">
+            <div className="flex w-full">
+              <input
+                className="font-space-mono bg-transparent outline-none text-white border-dotted border-white border-2 p-1 w-full text-lg"
+                type="text"
+                placeholder="Enter your code"
+                value={userPrivateKey}
+                onChange={(e) => setPrivateKey(e.target.value)}
+              />
+            </div>
+            <div className="flex w-full mt-4">
+              <BoxButton full size='small' style='outline' disabled={!userPrivateKey || !checkValidKey()} onClick={loadProgress} classes={`border-white border-2 p-1 text-xl md:w-full ${!userPrivateKey && 'opacity-50'}`}>Load my progress</BoxButton>
+            </div>
+            <div className="flex w-full mt-4 mb-4">
+              {userPrivateKey && !checkValidKey() && (
+                <div className='flex items-center'>
+                  <Warning />
```

Comment:
`warning.svg` should use `currentColor` for color definitions, so the value can be overwritten with CSS. Makes the icons more flexible.

Reply:
Done